### PR TITLE
fix(lightning): replace bincode encoding impl

### PR DESCRIPTION
**Pull request by elsirion**
Gets rid of one other place where we used `bincode` so far, follow up to #2296.

### CODE REVIEW

File: fedimint-core/src/encoding/tbs.rs [hunk 1]
```diff
@@ -97,6 +97,42 @@ impl Decodable for tbs::BlindingKey {
     }
 }
 
+impl Encodable for threshold_crypto::Ciphertext {
+    fn consensus_encode<W: Write>(&self, writer: &mut W) -> Result<usize, Error> {
+        self.to_bytes().consensus_encode(writer)
+    }
+}
+
+impl Decodable for threshold_crypto::Ciphertext {
```
Comment (m1sterc001guy) [inline fedimint-core/src/encoding/tbs.rs:106]: Just FYI, I needed to do something similar for `threshold_crypto::PublicKey` in https://github.com/fedimint/fedimint/pull/2279

**Comment by elsirion** [comment]
I think this will be the first time for me writing a DB migration judging by the CI failure ^^

**Comment by elsirion** [comment]
300 loc of DB migration code later … :laughing:

File: modules/fedimint-ln-server/src/lib.rs [hunk 2]
```diff
@@ -1160,3 +1170,264 @@ mod fedimint_migration_tests {
...
+                    expiry_time: value.expiry_time,
+                };
+
+                (new_key, new_value)
+            })
+            .collect::<Vec<_>>()
+            .await;
+
+        for (key, value) in offers {
+            // TODO: this overwrites, allow writing without reading without the warnings
```
Comment (m1sterc001guy) [inline modules/fedimint-ln-server/src/lib.rs:1356]: Confused by this comment

File: modules/fedimint-ln-server/src/lib.rs [hunk 3]
```diff
@@ -1160,3 +1170,264 @@ mod fedimint_migration_tests {
...
+        PreimageDecryptionShare,
+    };
+    use fedimint_ln_common::db::{
+        AgreedDecryptionShareKey, ContractKey, OfferKey, ProposeDecryptionShareKey,
+    };
+    use fedimint_ln_common::ContractAccount;
+    use futures::stream::StreamExt;
+    use serde::{Deserialize, Serialize};
+
+    pub async fn migration_v0(dbtx: &mut DatabaseTransaction<'_>) -> anyhow::Result<()> {
```
Comment (m1sterc001guy) [inline modules/fedimint-ln-server/src/lib.rs:1195]: Let's add a comment here on what the db migration is doing (i.e which structs changed encodings). This is to remove the dependency on `bincode` right?

**Comment by m1sterc001guy** [comment]
Do you think it is worth adding and maintaining this migration code? It might be worth taking advantage of the fact that we're pre-release and can make breaking changes for now :)

If we do want to go forward with it, can you add another backup in `prepare_migration_snapshots`? That way we'll be able to detect a db break if these structs ever change again.

**Comment by elsirion** [comment]
It was mostly a learning exercise for me to see how it would work. Not sure if worth keeping as a reference or if I should remove it. How would one go about simply re-generating the checked-in DB?

**Comment by dpc** [comment]
@elsirion This still alive? Should we just regen db and land?

**Comment by elsirion** [comment]
Yeah, regen+land, I'll take a look later.

**Comment by elsirion** [comment]
> Yeah, regen+land, I'll take a look later.

maxresdefault

**Review by Maan2003 — APPROVED** [review]
LGTM

**Review by Maan2003 — APPROVED** [review]

**Comment by Maan2003** [comment]
reviews from two tabs 😄

**Pull request merged**